src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -17,10 +17,227 @@ namespace System
     /// </summary>
     public static partial class MemoryExtensions
     {
+        /// <summary>
+        /// Moves the source characters to the destination with a specified number of characters starting at a specified position removed.
+        /// </summary>
+        /// <param name="source">The source span from which the characters are removed.</param>
+        /// <param name="startIndex">The zero-based position to begin deleting characters.</param>
+        /// <param name="count">The number of characters to delete.</param>
+        /// <param name="destination">The span to copy the leftover characters into.</param>
+        /// <exception cref="System.ArgumentException">
+        /// Thrown when the destination Span is too small to contain the source Span with the characters removed
+        /// (i.e. destination.Length &lt; source.Length - count).
+        /// </exception>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when either <paramref name="startIndex"/> or <paramref name="count"/> is less than zero or
+        /// when <paramref name="startIndex"/> plus <paramref name="count"/> specify a position outside the source span.
+        /// </exception>
```

Comment:
Could you add remark to the dcos saying that it's ok for destination and span to be overlapping (including pointing to the same memory)